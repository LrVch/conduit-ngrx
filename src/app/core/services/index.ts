export * from './api.service';
export * from './articles.service';
export * from './auth-guard.service';
export * from './comments.service';
export * from './profiles.service';
export * from './tags.service';
export * from './user.service';
export * from './custom-event-manager';
export * from './main-loader.service';
export * from './scroll.service';
export * from './dom-util.service';
export * from './local-storage.service';
export * from './animations.service';
