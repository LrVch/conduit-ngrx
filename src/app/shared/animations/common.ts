import {
  animate,
  query,
  style,
  transition,
  trigger,
  stagger,
  sequence,
  animation,
  group,
  useAnimation,
  state,
  animateChild
} from '@angular/animations';
import {
  AnimationsService,
  RouteAnimationChangeType
} from '@app/core/services/animations.service';

export const scaleOut = animation(
  [
    style({ opacity: 0, width: 0, 'min-width': 0, transform: 'scale(0,1)' }),
    group([
      animate('{{timeShort}}', style({ width: '*' })),
      animate('{{timeLong}}', style({ opacity: 1, transform: 'scale(1,1)' }))
    ])
  ],
  {
    params: {
      timeShort: '100ms ease-out',
      timeLong: '200ms ease-out'
    }
  }
);

export const scaleIn = animation(
  [
    style({ 'min-width': 0 }),
    group([
      animate('{{timeShort}}', style({ opacity: 0, transform: 'scale(0,1)' })),
      animate('{{timeLong}}', style({ width: 0 }))
    ])
  ],
  {
    params: {
      timeShort: '100ms ease-out',
      timeLong: '200ms ease-out'
    }
  }
);

export const expand = animation(
  [
    sequence([
      style({ opacity: 0, height: 0 }),
      animate('{{duration}} {{timing}}', style({ height: '*' })),
      animate('{{duration}} {{timing}}', style({ opacity: 1 }))
    ])
  ],
  {
    params: {
      duration: '300ms',
      timing: 'ease-out'
    }
  }
);

export const collapce = animation(
  [
    sequence([
      animate('{{duration}}', style({ opacity: 0 })),
      animate('{{duration}} {{timing}}', style({ height: 0 }))
    ])
  ],
  {
    params: {
      duration: '200ms',
      timing: 'ease-out'
    }
  }
);

export const avatarInOut = trigger('avatarInOut', [
  transition(':enter', [useAnimation(scaleOut)]),
  transition(':leave', [useAnimation(scaleIn)])
]);

export const commentAnimation = trigger('commentAnimation', [
  transition(':enter', [useAnimation(expand)]),
  transition(':leave', [useAnimation(collapce)])
]);

export const swingChange = animation(
  [
    sequence([
      query(':enter', [style({ opacity: 0, position: 'absolute' })]),
      query(':leave', [
        style({ transform: '{{transform}}(0)' }),
        animate(
          '{{duration}} {{timing}}',
          style({ opacity: 0, transform: '{{transform}}(100%)' })
        )
      ]),
      query(':leave', [style({ position: 'absolute' })]),
      query(':enter', [
        style({
          opacity: 0,
          transform: '{{transform}}({{direction}}100%)',
          position: 'static'
        }),
        animate(
          '{{duration}} {{timing}}',
          style({ opacity: 1, transform: '{{transform}}(0)' })
        )
      ])
    ])
  ],
  {
    params: {
      duration: '150ms',
      timing: 'cubic-bezier(0.35, 0, 0.25, 1)',
      transform: 'translateY',
      direction: '-'
    }
  }
);

export const deleteButtonAnimation = trigger('deleteButtonAnimation', [
  transition('false <=> true', [useAnimation(swingChange)])
]);

export const slidePageAnimation = animation(
  [
    style({ position: 'relative', overflow: 'hidden' }),
    query(':enter > *', [style({ opacity: 0, position: 'absolute' })], {
      optional: true
    }),
    query(':leave', animateChild(), {
      optional: true
    }),
    sequence([
      query(
        ':leave > *',
        [
          style({
            transform: 'translateX(0)',
            opacity: 1,
            position: 'relative'
          }),
          animate(
            '{{duration}} {{timing}}',
            style({ transform: 'translateX({{distanse}})', opacity: 0 })
          )
        ],
        {
          optional: true
        }
      ),
      query(':leave > *', style({ position: 'absolute' }), {
        optional: true
      }),
      query(
        ':enter > *',
        [
          style({
            position: 'relative',
            transform: 'translateX(-{{distanse}})'
          }),
          animate(
            '{{duration}} {{delay}} {{timing}}',
            style({ transform: 'translateX(0)', opacity: 1 })
          )
        ],
        {
          optional: true
        }
      )
    ]),
    query(':enter', animateChild(), {
      optional: true
    })
  ],
  {
    params: {
      distanse: '30px',
      duration: '300ms',
      delay: '300ms',
      timing: 'ease-out'
    }
  }
);

export const fadePageAnimation = animation(
  [
    style({ position: 'relative', overflow: 'hidden' }),
    query(':enter > *', [style({ opacity: 0, position: 'absolute' })], {
      optional: true
    }),
    query(':leave', animateChild(), {
      optional: true
    }),
    sequence([
      query(
        ':leave > *',
        [
          style({ opacity: 1, position: 'relative' }),
          animate('{{duration}} {{timing}}', style({ opacity: 0 }))
        ],
        {
          optional: true
        }
      ),
      query(':leave > *', style({ position: 'absolute' }), {
        optional: true
      }),
      query(
        ':enter > *',
        [
          style({ position: 'relative' }),
          animate('{{duration}} {{timing}}', style({ opacity: 1 }))
        ],
        {
          optional: true
        }
      )
    ]),
    query(':enter', animateChild(), {
      optional: true
    })
  ],
  {
    params: {
      distanse: '30px',
      duration: '300ms',
      timing: 'ease-out'
    }
  }
);

export function staticPages(fromState: string, toState: string) {
  return (
    toState === 'SettingsComponent' ||
    toState === 'EditorComponent' ||
    toState === 'login' ||
    toState === 'register'
  );
}

export function slideAnimation(fromState, toState) {
  if (!AnimationsService.isRouteAnimationsEnabled()) {
    return false;
  }

  if (!AnimationsService.isRouteAnimationsChangeType('SLIDE')) {
    return false;
  }

  return staticPages(fromState, toState);
}

export function fadeAnimation(fromState, toState) {
  if (!AnimationsService.isRouteAnimationsEnabled()) {
    return false;
  }

  if (!AnimationsService.isRouteAnimationsChangeType('FADE')) {
    return false;
  }

  return staticPages(fromState, toState);
}

export function typeAnimation(type: RouteAnimationChangeType, pages) {
  return state;
}

export const routeAnimation = trigger('routeAnimation', [
  transition(slideAnimation, [useAnimation(slidePageAnimation)]),
  transition(fadeAnimation, [useAnimation(fadePageAnimation)])
]);
