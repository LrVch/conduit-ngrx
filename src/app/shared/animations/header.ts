import {
    animate,
    query,
    style,
    transition,
    trigger,
    stagger,
    sequence
} from '@angular/animations';

export const a = '';
