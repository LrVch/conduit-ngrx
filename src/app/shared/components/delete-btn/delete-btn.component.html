<div *ngIf="showConfirm">
  <div [@deleteButtonAnimationY]="isOpenConfirm && type === 'y'"
       [@deleteButtonAnimationX]="isOpenConfirm && type === 'x'"
       class="delete__confirm-cont"
       [style.overflow]="type === 'x' ? 'hidden' : ''">
    <div *ngIf="isOpenConfirm && !isDeleting">
      <div class="delete__confirm">
        <mat-icon class="color-success"
                  (click)="onDelete(); isOpenConfirm = false"
                  aria-label="done">done</mat-icon>
        <mat-icon color="warn"
                  (click)="isOpenConfirm = false"
                  aria-label="close">close</mat-icon>
      </div>
    </div>
    <div *ngIf="!isOpenConfirm && !isDeleting">
      <mat-icon color="primary"
                (click)="isOpenConfirm = true"
                aria-label="delete">delete</mat-icon>
    </div>
    <div *ngIf="!isOpenConfirm && isDeleting">
      <mat-icon class="delete__remove-icon"
                (click)="isOpenConfirm = true"
                aria-label="delete_forever">delete_forever</mat-icon>
    </div>
  </div>
</div>

<div *ngIf="!showConfirm">
  <div [@deleteButtonAnimationY]="isDeleting && type === 'y' ? true : false"
       [@deleteButtonAnimationX]="isDeleting && type === 'x' ? true : false">
    <mat-icon *ngIf="!isDeleting"
              color="primary"
              [ngStyle]="{'opacity': isDeleting ? 0.5 : 1}"
              (click)="onDelete()"
              aria-label="delete">delete</mat-icon>
    <mat-icon *ngIf="isDeleting"
              class="delete__remove-icon"
              (click)="isOpenConfirm = true"
              aria-label="delete_forever">delete_forever</mat-icon>
  </div>
</div>