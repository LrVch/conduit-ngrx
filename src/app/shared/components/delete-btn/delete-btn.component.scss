:host {
    display: inline-block;
    vertical-align: middle;
    user-select: none;
}

.delete__confirm-cont {
    position: relative;
    white-space: nowrap;
}

.delete__remove-icon {
    color: red !important;
}