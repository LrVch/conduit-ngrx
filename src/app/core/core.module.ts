import { NgModule, Optional, SkipSelf } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { HttpTokenInterceptor } from './interceptors/http.token.interceptor';

import {
  ApiService,
  ArticlesService,
  AuthGuard,
  CommentsService,
  JwtService,
  ProfilesService,
  TagsService,
  UserService,
  CustomEventManager,
  ScrollService,
  DomUtilService
} from './services';
import { EventManager } from '@angular/platform-browser';

@NgModule({
  imports: [
    CommonModule
  ],
  providers: [
    { provide: HTTP_INTERCEPTORS, useClass: HttpTokenInterceptor, multi: true },
    ApiService,
    ArticlesService,
    AuthGuard,
    CommentsService,
    JwtService,
    ProfilesService,
    TagsService,
    UserService,
    ScrollService,
    DomUtilService,
    { provide: EventManager, useClass: CustomEventManager }
  ],
  declarations: []
})
export class CoreModule {
  constructor (@Optional() @SkipSelf() parentModule: CoreModule) {
    if (parentModule) {
      throw new Error(
        'CoreModule is already loaded. Import it in the AppModule only');
    }
  }
 }
