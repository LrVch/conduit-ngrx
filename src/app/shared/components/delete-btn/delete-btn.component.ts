import { Component, Input, Output, EventEmitter } from '@angular/core';
import { deleteButtonAnimationY, deleteButtonAnimationX } from '@app/shared/animations';

type AnimationType = 'x' | 'y';

@Component({
  selector: 'app-delete-btn',
  templateUrl: './delete-btn.component.html',
  styleUrls: ['./delete-btn.component.scss'],
  animations: [
    deleteButtonAnimationY,
    deleteButtonAnimationX
  ]
})
export class DeleteBtnComponent  {
  @Input('type') type: AnimationType = 'y';
  @Input('showConfirm') showConfirm = true;
  @Input('isDeleting') isDeleting = false;
  @Output() delete = new EventEmitter<null>();

  isOpenConfirm = false;

  onDelete() {
    this.delete.emit();
  }
}
