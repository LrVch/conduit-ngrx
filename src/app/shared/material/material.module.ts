import { NgModule } from '@angular/core';
import {
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
} from '@angular/material';

const modules = [
  MatButtonModule,
  MatCheckboxModule,
  MatToolbarModule,
  MatCardModule,
  MatChipsModule,
  MatProgressBarModule,
  MatPaginatorModule,
  MatTabsModule,
  MatIconModule,
  MatInputModule,
  MatProgressSpinnerModule,
  MatDialogModule,
  MatSnackBarModule
];

@NgModule({
  imports: modules,
  exports: modules
})
export class MaterialModule { }
