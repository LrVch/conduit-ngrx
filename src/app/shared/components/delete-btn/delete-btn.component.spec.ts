import { async, ComponentFixture } from '@angular/core/testing';

import { ConfigureFn, configureTests } from '@app/lib/testing';

import { DeleteBtnComponent } from './delete-btn.component';
import { DebugElement } from '@angular/core';
import { By } from '@angular/platform-browser';
import { MaterialModule } from '@app/shared/material/material.module';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';

describe('DeleteBtnComponent', () => {
  let component: DeleteBtnComponent;
  let fixture: ComponentFixture<DeleteBtnComponent>;
  let de: DebugElement;
  let el: HTMLElement;
  const event = {
    prevented: false,
    preventDefault() {
      this.prevented = true;
    }
  };

  beforeEach(async(() => {
    const configure: ConfigureFn = testBed => {
      testBed.configureTestingModule({
        declarations: [DeleteBtnComponent],
        imports: [MaterialModule, NoopAnimationsModule]
      });
    };

    configureTests(configure).then(testBed => {
      fixture = testBed.createComponent(DeleteBtnComponent);
      component = fixture.componentInstance;
      de = fixture.debugElement;
      el = de.nativeElement;
      fixture.detectChanges();
    });
  }));

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should has "type" @Input equals to "y"', () => {
    expect(component.type).toBe('y');
  });

  it('should has "direction" @Input equals to "forward"', () => {
    expect(component.direction).toBe('forward');
  });

  it('should has "showConfirm" @Input"', () => {
    expect(component.showConfirm).toBe(true);
  });

  it('should has "isDeleting" @Input"', () => {
    expect(component.isDeleting).toBe(false);
  });

  it('should has "isOpenConfirm" @Input"', () => {
    expect(component.isOpenConfirm).toBe(false);
  });

  it('should compile showConfirm == true', () => {
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should compile showConfirm == false', () => {
    component.showConfirm = false;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should show variants for showConfirm == true', () => {
    component.isOpenConfirm = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should show delete_forever for showConfirm == true', () => {
    component.isOpenConfirm = false;
    component.isDeleting = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should show delete_forever for showConfirm == false', () => {
    component.showConfirm = false;
    component.isDeleting = true;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should raises delete event when clicked (onDelete) for showConfirm == true', () => {
    let raised = null;
    component.isOpenConfirm = true;
    component.delete.subscribe(_ => (raised = true));

    fixture.detectChanges();

    const button = de.query(By.css('[class*="color-success"]'));
    button.triggerEventHandler('click', event);

    expect(raised).toBe(true);
  });

  it('should raises delete event when clicked (onDelete) for showConfirm == false', () => {
    let raised = null;
    component.showConfirm = false;
    component.delete.subscribe(_ => (raised = true));

    fixture.detectChanges();

    const button = de.query(By.css('[color="primary"]'));
    button.triggerEventHandler('click', event);

    expect(raised).toBe(true);
  });
});
