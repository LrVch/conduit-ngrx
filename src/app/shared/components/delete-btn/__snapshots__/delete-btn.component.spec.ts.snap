// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteBtnComponent should compile showConfirm == false 1`] = `
<app-delete-btn
  delete={[Function EventEmitter]}
  direction={[Function String]}
  isDeleting="false"
  isOpenConfirm="false"
  showConfirm="false"
  type={[Function String]}
>
  <div
    class="ng-tns-c14-0 ng-star-inserted"
  >
    <div
      class="ng-tns-c14-0 ng-trigger ng-trigger-deleteButtonAnimation"
      style=""
    >
      
      <mat-icon
        aria-hidden="true"
        aria-label="delete"
        class="mat-icon ng-tns-c14-0 mat-primary material-icons ng-star-inserted"
        color="primary"
        ng-reflect-color="primary"
        ng-reflect-ng-style="[object Object]"
        role="img"
        style="opacity: 1;"
      >
        delete
      </mat-icon>
      
    </div>
  </div>
</app-delete-btn>
`;

exports[`DeleteBtnComponent should compile showConfirm == true 1`] = `
<app-delete-btn
  delete={[Function EventEmitter]}
  direction={[Function String]}
  isDeleting="false"
  isOpenConfirm="false"
  showConfirm={[Function Boolean]}
  type={[Function String]}
>
  <div
    class="ng-tns-c12-0 ng-star-inserted"
  >
    <div
      class="delete__confirm-cont ng-trigger ng-trigger-deleteButtonAnimation"
      style=""
    >
      
      
      <div
        class="ng-tns-c12-0 ng-star-inserted"
        style=""
      >
        <mat-icon
          aria-hidden="true"
          aria-label="delete"
          class="mat-icon mat-primary material-icons"
          color="primary"
          ng-reflect-color="primary"
          role="img"
        >
          delete
        </mat-icon>
      </div>
      
    </div>
  </div>
</app-delete-btn>
`;

exports[`DeleteBtnComponent should show delete_forever for showConfirm == false 1`] = `
<app-delete-btn
  delete={[Function EventEmitter]}
  direction={[Function String]}
  isDeleting={[Function Boolean]}
  isOpenConfirm="false"
  showConfirm="false"
  type={[Function String]}
>
  <div
    class="ng-tns-c20-0 ng-star-inserted"
  >
    <div
      class="ng-tns-c20-0 ng-trigger ng-trigger-deleteButtonAnimation"
      style=""
    >
      
      
      <mat-icon
        aria-hidden="true"
        aria-label="delete_forever"
        class="delete__remove-icon mat-icon ng-tns-c20-0 material-icons ng-star-inserted"
        role="img"
        style=""
      >
        delete_forever
      </mat-icon>
    </div>
  </div>
</app-delete-btn>
`;

exports[`DeleteBtnComponent should show delete_forever for showConfirm == true 1`] = `
<app-delete-btn
  delete={[Function EventEmitter]}
  direction={[Function String]}
  isDeleting={[Function Boolean]}
  isOpenConfirm="false"
  showConfirm={[Function Boolean]}
  type={[Function String]}
>
  <div
    class="ng-tns-c18-0 ng-star-inserted"
  >
    <div
      class="delete__confirm-cont ng-trigger ng-trigger-deleteButtonAnimation"
      style=""
    >
      
      
      
      <div
        class="ng-tns-c18-0 ng-star-inserted"
        style=""
      >
        <mat-icon
          aria-hidden="true"
          aria-label="delete_forever"
          class="delete__remove-icon mat-icon material-icons"
          role="img"
        >
          delete_forever
        </mat-icon>
      </div>
    </div>
  </div>
</app-delete-btn>
`;

exports[`DeleteBtnComponent should show variants for showConfirm == true 1`] = `
<app-delete-btn
  delete={[Function EventEmitter]}
  direction={[Function String]}
  isDeleting="false"
  isOpenConfirm={[Function Boolean]}
  showConfirm={[Function Boolean]}
  type={[Function String]}
>
  <div
    class="ng-tns-c16-0 ng-star-inserted"
  >
    <div
      class="delete__confirm-cont ng-trigger ng-trigger-deleteButtonAnimation"
      style=""
    >
      
      <div
        class="ng-tns-c16-0 ng-star-inserted ng-animating"
        style=""
      >
        <div
          class="delete__confirm"
        >
          <mat-icon
            aria-hidden="true"
            aria-label="done"
            class="color-success mat-icon material-icons"
            role="img"
          >
            done
          </mat-icon>
          <mat-icon
            aria-hidden="true"
            aria-label="close"
            class="mat-icon mat-warn material-icons"
            color="warn"
            ng-reflect-color="warn"
            role="img"
          >
            close
          </mat-icon>
        </div>
      </div>
      
      <div
        class="ng-tns-c16-0 ng-star-inserted ng-animating"
        style=""
      >
        <mat-icon
          aria-hidden="true"
          aria-label="delete"
          class="mat-icon mat-primary material-icons"
          color="primary"
          ng-reflect-color="primary"
          role="img"
        >
          delete
        </mat-icon>
      </div>
      
    </div>
  </div>
</app-delete-btn>
`;
